diff --git a/python/sglang/srt/distributed/device_communicators/custom_all_reduce.py b/python/sglang/srt/distributed/device_communicators/custom_all_reduce.py
index 6e7ea07e7..73b512f51 100644
--- a/python/sglang/srt/distributed/device_communicators/custom_all_reduce.py
+++ b/python/sglang/srt/distributed/device_communicators/custom_all_reduce.py
@@ -64,6 +64,7 @@ class CustomAllreduce:
         group: ProcessGroup,
         device: Union[int, str, torch.device],
         max_size=_MAX_CAR_SIZE,
+        enable_register_for_capturing: bool = True,
     ) -> None:
         """
         Args:
@@ -410,6 +411,8 @@ class CustomAllreduce:
         if self._IS_CAPTURING:
             if torch.cuda.is_current_stream_capturing():
                 if _is_hip:
+                    if self.tms_cudagraph:
+                        return self.all_reduce_unreg(input)
                     return self.all_reduce_reg(input)
                 else:
                     return self.all_reduce(input, registered=not self.tms_cudagraph)
diff --git a/python/sglang/srt/distributed/parallel_state.py b/python/sglang/srt/distributed/parallel_state.py
index c3ca1e4f3..2bb763b6a 100644
--- a/python/sglang/srt/distributed/parallel_state.py
+++ b/python/sglang/srt/distributed/parallel_state.py
@@ -351,10 +351,12 @@ class GroupCoordinator:
         if use_custom_allreduce and self.world_size > 1:
             # Initialize a custom fast all-reduce implementation.
             try:
+                tms_cudagraph = envs.SGLANG_MEMORY_SAVER_CUDA_GRAPH.get()
                 CAClass = dispatch_custom_allreduce()
                 self.ca_comm = CAClass(
                     group=self.cpu_group,
                     device=self.device,
+                    enable_register_for_capturing=not tms_cudagraph,
                 )
             except Exception as e:
                 logger.warning(
